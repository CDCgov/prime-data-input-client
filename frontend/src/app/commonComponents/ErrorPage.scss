.error-img-container{ 
    width: 12rem !important;
    margin: 1.5rem;

}

.error-h1 {
    margin-bottom: 2rem;
}

.error-text-container {
    text-align: center;
}