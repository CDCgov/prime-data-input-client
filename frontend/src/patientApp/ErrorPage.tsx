import { ApolloError } from "@apollo/client";

interface Props {
  error: ApolloError;
}

const ErrorPage: any = (props: Props) => {
  return (
    <main className="prime-home">
      <div className="grid-container">
        <div className="grid-row">
          <div className="prime-container usa-card__container">
            <div className="usa-card__header">
              <div>
                <h2>Error</h2>
              </div>
            </div>
            <div className="usa-card__body">{props.error.message}</div>
          </div>
        </div>
      </div>
    </main>
  );
};

export default ErrorPage;
