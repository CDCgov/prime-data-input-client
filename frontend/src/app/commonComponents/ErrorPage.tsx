import siteLogo from "../../img/simplereport-logo-color.svg";
import "./ErrorPage.scss";

const ErrorPage = () => {
    return (
      <main>
        <div className="error-img-container">
            <div className="usa-logo">
                <img src={siteLogo}
                alt = "{process.env.REACT_APP_TITLE}"/>
            </div>
        </div>
        <div className="grid-container maxw-tablet error-text-container">
          <h1 className="error-h1">Something went wrong :(</h1>
          <p>
            Please try refreshing your browser.
          </p>
          <p>If the problem continues, contact support@simplereport.gov for support.</p>
        </div>
      </main>
    );
  };
  
  export default ErrorPage;