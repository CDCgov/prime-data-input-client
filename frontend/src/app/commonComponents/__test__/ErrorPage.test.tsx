import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import ErrorPage from "../ErrorPage";

describe("ErrorPage", () => {
  it("displays the correct error message", () => {
    const { container, getByText } = render(
      <MemoryRouter>
        <ErrorPage />
      </MemoryRouter>
    );
    expect(getByText("Something went wrong :(")).toBeInTheDocument();
    expect(
      getByText("Please try refreshing your browser.")
    ).toBeInTheDocument();
    expect(getByText("support@simplereport.gov")).toHaveAttribute('href', 'mailto:support@simplereport.gov');
    expect(container).toMatchSnapshot();
  });
});
