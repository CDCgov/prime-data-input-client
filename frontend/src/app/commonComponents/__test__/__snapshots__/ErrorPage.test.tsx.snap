// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage displays the correct error message 1`] = `
<div>
  <main>
    <div
      class="error-img-container"
    >
      <div
        class="usa-logo"
      >
        <img
          alt="{process.env.REACT_APP_TITLE}"
          src="simplereport-logo-color.svg"
        />
      </div>
    </div>
    <div
      class="grid-container maxw-tablet error-text-container"
    >
      <h1
        class="error-h1"
      >
        Something went wrong :(
      </h1>
      <p>
        Please try refreshing your browser.
      </p>
      <p>
        If the problem continues, contact support@simplereport.gov for support.
      </p>
    </div>
  </main>
</div>
`;
